pipenv/cli.py

```diff
@@ -105,11 +105,16 @@ def ensure_pipfile(validate=True):
             # Pip requires a `PipSession` which is a subclass of requests.Session.
             # Since we're not making any network calls, it's initialized to nothing.
             from pip.req.req_file import parse_requirements
-            reqs = [r.req for r in parse_requirements(project.requirements_location, session='')]
+            reqs = [r for r in parse_requirements(project.requirements_location, session='')]
 
             for package in reqs:
                 if package.name not in BAD_PACKAGES:
-                    project.add_package_to_pipfile(str(package))
+                    if package.original_link is not None:
+                        pack_link = str(package.original_link)
```

Comment:
Looked into it.  `original_link`  and `link` is of type `pip.index.Link`.  I changed up the logic a bit to remove making it a string, since .format will take care of it in one of the scenarios.